src/validation/data/package.rs

```diff
@@ -61,13 +61,23 @@ impl Package {
     /// Try to create a new one from a [`FileDescriptorProto`].
     pub fn new(descriptor: &FileDescriptorProto) -> Result<Self, PackageError> {
         let mut package = Self {
-            file: descriptor.name().into(),
+            files: vec![descriptor.name().into()],
             name: descriptor.package().to_string(),
             entities: Default::default(),
         };
 
+        package.parse(descriptor)?;
+
+        Ok(package)
+    }
+
+    pub fn add(&mut self, descriptor: &FileDescriptorProto) {
+        self.parse(descriptor).unwrap();
+    }
+
+    fn parse(&mut self, descriptor: &FileDescriptorProto) -> Result<&Self, PackageError> {
```

Comment:
I think you can inline it in here @JoeHut – so taking a slice of file descriptors, instead of a single one? @xfbs what do you think?

Reply:
also it is good practice to reduce the number of functions that take an `&mut self` to a minimum since they encourage imperative programing. 

This is essentially a parsing problem which can be solved nicely in a functional manner – so to say "deriving" the contents from the file descriptor types.

So what you want to do (i think):

```haskell
type PackageDescriptors = [FileDescriptor]

parse :: PackageDescriptors -> Package
```

– 

translating this into rust:

```
struct DirectoryDescriptorSet<'a>(&'a [FileDescriptorProto]);

impl<'a> DirectoryDescriptorSet<'a> {
    // implement a mapping function that ensures that the `FileDescriptorProto` structs are located next to eachother in the typesystem
}

struct Package { ... }

impl<'a> TryFrom<DirectoryDescriptorSet<'a>> for Package {
    type Err = miette::Report;

    // implement everything to do type conversion from A (DirectoryDescriptorSet) to B (Package)
    // fn try_from ...
}
```

This way the flow of the program looks like:

1. Collect file descriptors grouped by directory
2. Create DirectoryDescriptorSets from them
3. Try to convert them to one Package using `TryFrom`

cc @xfbs